### PR TITLE
fix: CodeEditor issues on resizing

### LINKED ISSUE #23796: [Bug]: Code selection is broken when the width of property pane is larger than default
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Code selection feature breaks when a property pane is resized to a larger width.

Happening on release.app.appsmith.com, not on prod. May be related to ADS 2.0 changes

### Steps To Reproduce

1. DnD a chart widget
2. Choose Custom Chart
3. Select/Double click on any parameter passed in the property. Notice that it is selectable.
4. Increase the width of property pane.
5. Try to select/double click on any value mentioned in the pane. Notice that it is not selectable.

Expected behavior : It should be selectable.

### Public Sample App

_No response_

### Environment

Production

### Issue video log

https://www.loom.com/share/a20bb0fb5e7a4b859f6fb0ebf65286c3

### Version

Cloud/ADS 2.0

### PR DESCRIPTION
**Pull request by aswathkk**
## Description
CodeMirror code editor is not being refreshed when we resize it. This caused issue #23796

- Added a ResizeObserver which will call `editor.refresh()` on resize
- Removed `containerHeight` props which is no longer useful due to the introduction of the resize observer. This was introduced in #16971

#### PR fixes following issue(s)
Fixes #23796

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by hetunandu — APPROVED** [review]

File: app/client/src/utils/resizeObserver.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    callbacks.push(callback);
+    this.callbacksMap.set(target, [callback]);
+  }
+
+  /**
+   *
+   * @param target The element to unobserve
+   * @param callback The callback to be removed
+   */
+  unobserve(target: Element, callback: ResizeObserCallback) {
```
Comment (arunvjn) [inline app/client/src/utils/resizeObserver.ts:46]: Minor nit - callbacks could be an array here. Allows for easier customisation

**Review by arunvjn — DISMISSED** [review]

File: app/client/src/utils/resizeObserver.ts (hunk 1 above)
Comment (aswathkk) [reply app/client/src/utils/resizeObserver.ts:46]: What you mean by easier customisation?

Can you give me an example?

**Review by arunvjn — APPROVED** [review]

**Comment by kamakshibhat-appsmith** [comment]
@chandannkumar Can you pls QA this ?

**Comment by chandannkumar** [comment]
Tested this PR and LGTM

**Review by arunvjn — APPROVED** [review]

**Pull request merged**